hooks/tk-multi-publish2/basic/publish_renders.py

```diff
@@ -0,0 +1,510 @@
+                enabled_flags = [x.getValue(0) for x in outputs.getChild("enabledFlags").getChildren()]
+                for path, enabled in zip(locations, enabled_flags):
+                    render_paths[path] |= bool(enabled)  # At least one enabled to make it enabled
+            self.__render_paths = dict(render_paths)
+        return self.__render_paths
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+        node = item.properties["node"]
+        settings["node"].value = node.getName()
+        path = item.properties["path"]
+        work_template = item.properties["work_template"]
+        publish_template = item.properties["publish_template"]
+
+        if not work_template or not publish_template:
+            self.logger.debug("Templates do not exist. Is the context set propertly?")
+            return {"accepted": False, "visible": False, "enabled": False, "checked": False}
+
+        fields = work_template.validate_and_get_fields(path)
+        if not fields:
+            self.logger.debug(
+                "'{}': '{}', is not a valid path".format(node_name, path)
+            )
+            return {"accepted": False, "visible": False, "enabled": False, "checked": False}
+        del fields["version"]
+
+        work_paths = self.sgtk.abstract_paths_from_template(work_template, fields)
+        if not work_paths:
+            self.logger.debug(
+                "'{}': No generated work files".format(node_name)
+            )
+            return {"accepted": False, "visible": True, "enabled": True, "checked": False}
+
+        publish_paths = self.sgtk.abstract_paths_from_template(publish_template, fields)
+
+        work_paths_to_publish = sorted(work_paths, reverse=True)
+        publish_fields = {}
```

Comment:
Could this be made into `self._published_fields = defaultdict(dict)` and cache `publish_template` and `publish_path` together. e.g. at line 331
```python
self._published_fields[publish_template][publish_path] = publish_template.get_fields(publish_path)
```